drivers/i3c/i3c_cdns.c

```diff
@@ -2387,6 +2387,102 @@ static void cdns_i3c_target_ibi_hj_complete(const struct device *dev)
 }
 #endif
 
+static void cdns_i3c_slv_sdr_tx_th_int_handler(const struct device *dev,
+					       const struct i3c_target_callbacks *target_cb)
+{
+	int status = 0;
+	struct cdns_i3c_data *data = dev->data;
+	const struct cdns_i3c_config *config = dev->config;
+
+	if (target_cb != NULL && target_cb->read_processed_cb) {
+		/* while tx fifo is not full and there is still data available */
+		while ((!(sys_read32(config->base + SLV_STATUS1) & SLV_STATUS1_SDR_TX_FULL)) &&
+		       (status == 0)) {
+			/* call function pointer for read */
+			/* with rev 1p7, as a target, the fifos are full word,
+			 * otherwise only the first byte is used
+			 */
+			if (REV_ID_REV(data->hw_cfg.rev_id) >= REV_ID_VERSION(1, 7)) {
+				uint32_t tx_data = 0;
+				bool data_valid = false;
+
+				for (int j = 0; j < 4; j++) {
+					uint8_t byte;
+					/* will return negative if no data left to transmit and 0
+					 * if data available
+					 */
+					status = target_cb->read_processed_cb(data->target_config,
+									      &byte);
+					if (status == 0) {
+						data_valid = true;
+						tx_data |= (byte << (j * 8));
+					}
+				}
+				if (data_valid) {
+					cdns_i3c_write_tx_fifo(config, &tx_data, sizeof(uint32_t));
+				}
+			} else {
+				uint8_t byte;
+				/* will return negative if no data left to transmit and 0 if
+				 * data available
+				 */
+				status = target_cb->read_processed_cb(data->target_config, &byte);
+				if (status == 0) {
+					cdns_i3c_write_tx_fifo(config, &byte, sizeof(uint8_t));
+				}
+			}
+		}
+	}
+}
+
+static void cdns_i3c_slv_ddr_tx_th_int_handler(const struct device *dev,
+					       const struct i3c_target_callbacks *target_cb)
+{
+	int status = 0;
+	struct cdns_i3c_data *data = dev->data;
+	const struct cdns_i3c_config *config = dev->config;
+
+	if (target_cb != NULL && target_cb->read_processed_cb) {
+		while ((!(sys_read32(config->base + SLV_STATUS1) & SLV_STATUS1_DDR_TX_FULL)) &&
+		       (status == 0)) {
+			/* call function pointer for read */
+			/* with rev 1p7, as a target, the fifos are full word,
+			 * otherwise only the first byte is used
+			 */
+			if (REV_ID_REV(data->hw_cfg.rev_id) >= REV_ID_VERSION(1, 7)) {
+				uint32_t tx_data = 0;
+				bool data_valid = false;
+
+				for (int j = 0; j < 4; j++) {
+					uint8_t byte;
+					/* will return negative if no data left to transmit and 0
+					 * if data available
+					 */
+					status = target_cb->read_processed_cb(data->target_config,
+									      &byte);
+					if (status == 0) {
+						data_valid = true;
+						tx_data |= (byte << (j * 8));
+					}
+				}
+				if (data_valid) {
+					cdns_i3c_write_ddr_tx_fifo(config, &tx_data,
```

Comment:
I'm not so sure this will work (or was it even working before hand). This fifo is a 20b fifo that should contain the full ddr word (along with preamble, parity, 16b data)

Reply:
threshold interrupt for ddr slv is not supported yet based on the code comments, removing ddr related changes from this patch.